server/src/com/thoughtworks/go/config/update/AgentsEntityConfigUpdateCommand.java

```diff
@@ -60,55 +65,38 @@ public AgentsEntityConfigUpdateCommand(Username username, LocalizedOperationResu
 
     @Override
     public boolean canContinue(CruiseConfig cruiseConfig) {
-        if (goConfigService.isAdministrator(username.getUsername())) {
-            return true;
+        if (!isAuthorized()) {
+            return false;
         }
 
-        result.unauthorized(LocalizedMessage.string("UNAUTHORIZED_TO_OPERATE_AGENTS"), HealthStateType.unauthorised());
-        return false;
-    }
-
-    private List<AgentConfig> getValidAgents(List<String> uuids, LocalizedOperationResult result, Agents agents) throws NoSuchAgentException {
-        List<AgentConfig> goodAgents = new ArrayList<>();
-        List<String> unknownAgents = new ArrayList<>();
-
-        for (String uuid : uuids) {
-            AgentConfig agent = agents.getAgentByUuid(uuid);
-            if (!agent.isNull()) {
-                goodAgents.add(agent);
-            } else {
-                unknownAgents.add(uuid);
-            }
-        }
-        if (!unknownAgents.isEmpty()) {
-            result.badRequest(LocalizedMessage.string("AGENTS_WITH_UUIDS_NOT_FOUND", unknownAgents));
-            throw new NoSuchAgentException(unknownAgents);
+        if (isAnyOperationPerformedOnAgents()) {
+            return true;
         }
 
-        return goodAgents;
-    }
-
-    private void validateEnvironment(Set<CaseInsensitiveString> allEnvironmentNames, List<String> environmentsToOperate, LocalizedOperationResult result) throws NoSuchEnvironmentException {
-        for (String environment : environmentsToOperate) {
-            CaseInsensitiveString environmentName = new CaseInsensitiveString(environment);
-            if (!allEnvironmentNames.contains(environmentName)) {
-                result.badRequest(LocalizedMessage.string("RESOURCE_NOT_FOUND", "Environment", environmentName));
-                throw new NoSuchEnvironmentException(environmentName);
-            }
-        }
+        result.badRequest(LocalizedMessage.string("NO_OPERATION_PERFORMED_ON_AGENTS"));
+        return false;
     }
 
     @Override
     public void update(CruiseConfig preprocessedConfig) throws Exception {
-        List<AgentConfig> goodAgents = getValidAgents(uuids, result, preprocessedConfig.agents());
-        checkElasticAgentsResourceUpdated(goodAgents, resourcesToAdd, resourcesToRemove);
-
         Set<CaseInsensitiveString> allEnvironmentNames = new HashSet<>(goConfigService.getEnvironments().names());
 
-        validateEnvironment(allEnvironmentNames, environmentsToAdd, result);
-        validateEnvironment(allEnvironmentNames, environmentsToRemove, result);
+        // validate all inputs
+        validateEnvironmentNames(allEnvironmentNames, environmentsToAdd);
```

Comment:
Consider renaming the method, as you are just checking for the presense of the environment in the config.